pkg/k8srbac/subjects_test.go

```diff
@@ -0,0 +1,90 @@
+package k8srbac
+
+import (
+	"testing"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protoassert"
+)
+
+func TestGetSubjectsAdjustedByKind(t *testing.T) {
+	tests := map[string]struct {
+		rb     *storage.K8SRoleBinding
+		expect []*storage.Subject
+	}{
+		"nil rolebinding": {
+			rb:     nil,
+			expect: nil,
+		},
+		"nil subjects": {
+			rb:     &storage.K8SRoleBinding{},
+			expect: nil,
+		},
+		"empty subjects": {
+			rb:     &storage.K8SRoleBinding{Subjects: make([]*storage.Subject, 0)},
+			expect: nil,
+		},
+		"all kinds supported": {
+			rb: &storage.K8SRoleBinding{Subjects: []*storage.Subject{
+				{Kind: storage.SubjectKind_UNSET_KIND},
+				{Kind: storage.SubjectKind_SERVICE_ACCOUNT},
+				{Kind: storage.SubjectKind_USER},
+				{Kind: storage.SubjectKind_GROUP},
+			}},
+			expect: []*storage.Subject{
+				{Kind: storage.SubjectKind_UNSET_KIND},
+				{Kind: storage.SubjectKind_SERVICE_ACCOUNT},
+				{Kind: storage.SubjectKind_USER},
+				{Kind: storage.SubjectKind_GROUP},
+			},
+		},
+		"namespaced kinds preserve namespace": {
+			rb: &storage.K8SRoleBinding{Subjects: []*storage.Subject{
+				{Kind: storage.SubjectKind_UNSET_KIND, Namespace: "namespace"},
+				{Kind: storage.SubjectKind_SERVICE_ACCOUNT, Namespace: "namespace"},
+			}},
+			expect: []*storage.Subject{
+				{Kind: storage.SubjectKind_UNSET_KIND, Namespace: "namespace"},
+				{Kind: storage.SubjectKind_SERVICE_ACCOUNT, Namespace: "namespace"},
+			},
+		},
+		"non-namespaced kinds are adjusted": {
+			rb: &storage.K8SRoleBinding{Subjects: []*storage.Subject{
+				{Kind: storage.SubjectKind_USER, Namespace: "namespace"},
+				{Kind: storage.SubjectKind_GROUP, Namespace: "namespace"},
+			}},
+			expect: []*storage.Subject{
+				{Kind: storage.SubjectKind_USER},
+				{Kind: storage.SubjectKind_GROUP},
+			},
+		},
```

Comment:
nitpick: could you add a test where the input has a mix of namespaced and non-namespaced subjects (like the re-union of the above two tests)?